genie-web/src/main/java/com/netflix/genie/web/services/impl/JobResolverServiceImpl.java

```diff
@@ -696,30 +668,48 @@ private String tagsToString(final Set<String> tags) {
     /**
      * Helper to convert archive location prefix to an archive location.
      *
-     * @param requestedArchiveLocationPrefix archive location prefix uri
-     * @param jobId                          job id
-     * @return archive location for the job
+     * @param context The current resolution context
      */
-    private String toArchiveLocation(
-        final String requestedArchiveLocationPrefix,
-        final String jobId
-    ) {
+    private void resolveArchiveLocation(final JobResolutionContext context) {
+        // TODO: Disable ability to disable archival for all jobs during internal V4 migration.
+        //       Will allow us to reach out to clients who may set this variable but still expect output after
+        //       job completion due to it being served off the node after completion in V3 but now it won't.
+        //       Put this back in once all use cases have been hunted down and users are sure of their expected
+        //       behavior
+        final String requestedArchiveLocationPrefix =
+            context.getJobRequest()
+                .getRequestedJobArchivalData()
+                .getRequestedArchiveLocationPrefix()
+                .orElse(this.defaultArchiveLocation);
+        final String jobId = context.getJobId();
         final String archivePrefix = StringUtils.isBlank(requestedArchiveLocationPrefix)
             ? this.defaultArchiveLocation
             : requestedArchiveLocationPrefix;
 
-        if (archivePrefix.endsWith(File.separator)) {
-            return archivePrefix + jobId;
-        } else {
-            return archivePrefix + File.separator + jobId;
-        }
+        context.setArchiveLocation(
+            archivePrefix.endsWith(File.separator)
+                ? archivePrefix + jobId
+                : archivePrefix + File.separator + jobId
+        );
     }
 
-    private int resolveJobMemory(final JobRequest jobRequest, final Command command) {
-        return jobRequest
-            .getRequestedJobEnvironment()
-            .getRequestedJobMemory()
-            .orElse(command.getMemory().orElse(this.defaultMemory));
+    private void resolveJobMemory(final JobResolutionContext context) {
+        context.setJobMemory(
+            context.getJobRequest()
+                .getRequestedJobEnvironment()
+                .getRequestedJobMemory()
+                .orElse(
+                    context
+                        .getCommand()
+                        .orElseThrow(
+                            () -> new IllegalStateException(
```

Comment:
Seems to me like this and similar unchecked exception will bomb resolution, rather than throwing `GenieJobResolutionException`

Reply:
The top level public API implementations both have the following catch blocks:
```
        } catch (final GenieJobResolutionException e) {
            MetricsUtils.addFailureTagsWithException(tags, e);
            throw e;
        } catch (final Throwable t) {
            MetricsUtils.addFailureTagsWithException(tags, t);
            throw new GenieJobResolutionException(t);
        }
```
 ensuring the checked contract is held up outside the class implementation.